x-pack/plugins/ml/server/routes/README.md

```diff
@@ -6,11 +6,14 @@ Each route handler requires [apiDoc](https://github.com/apidoc/apidoc) annotatio
 to generate documentation.
 The [apidoc-markdown](https://github.com/rigwild/apidoc-markdown) package is also required in order to generate the markdown.
 
-For now the process is pretty manual. You need to make sure the packages mentioned above are installed globally 
-to execute the following command from the directory in which this README file is located.
+There are custom parser and worker (`x-pack/plugins/ml/server/routes/apidoc_scripts`) to process api schemas for each documentation entry. It's written with typescript so make sure all the scripts in the folder are compiled before executing `apidoc` command.
+
+Make sure you have run `yarn kbn bootstrap` to get all requires dev dependencies. Then execute the following command from the ml plugin folder: 
 ```
-apidoc -i . -o ../routes_doc && apidoc-markdown -p ../routes_doc -o ../routes_doc/ML_API.md 
+yarn run apiDocs
 ```
+It compiles all the required scripts and generates the documentation both in HTML and Markdown formats.
+
 
 It will create a new directory `routes_doc` (next to the `routes` folder) which contains the documentation in HTML format 
-as well as `ML_API.md` file.
\ No newline at end of file
+as well as markdown files under `/md` folder.
```

Comment:
Is it possible to create an `index.md` file, as otherwise we will need and maintain that.

Reply:
going to create a feature-request in the `apidoc-markdown` repo